DashboardClient/DashboardClient.cpp

```diff
@@ -27,6 +27,173 @@ namespace Umati
+							placeholderNodeUnconst->removeInstance(*it);
+							deleteAndUnsubscribeNode(*it);
+						}
+					}
+				}
+			}
+			//Hotfix clear subscription cache.
+			m_subscribedValues.clear();
+		}
+		void DashboardClient::deleteAndUnsubscribeNode(ModelOpcUa::PlaceholderElement placeHolderElement) {
+			std::shared_ptr<const ModelOpcUa::SimpleNode> element = placeHolderElement.pNode;
+			const std::list<std::shared_ptr<const ModelOpcUa::Node>> children = element->ChildNodes;
+			for(auto it = children.begin(); it != children.end();it++) {
+				std::shared_ptr<const ModelOpcUa::Node> node = *it;
+				std::shared_ptr<const ModelOpcUa::SimpleNode> simpleNode = std::dynamic_pointer_cast<const ModelOpcUa::SimpleNode>(node);
+				if(simpleNode != nullptr) {
+					const ModelOpcUa::SimpleNode simpleN = *simpleNode;
+					deleteAndUnsubscribeNode(simpleN);
+				} else {
+					std::shared_ptr<const ModelOpcUa::PlaceholderNode> placeHolderNode = std::dynamic_pointer_cast<const ModelOpcUa::PlaceholderNode>(node);
+					if(placeHolderNode != nullptr) {
+						std::list<ModelOpcUa::PlaceholderElement> instances = placeHolderNode->getInstances();
+						for(std::list<ModelOpcUa::PlaceholderElement>::iterator it = instances.begin(); it != instances.end(); it++) {
+							deleteAndUnsubscribeNode(*it);
+						}
+
+					}
+				}
+			}
+			LOG(INFO) << "Erase NodeId :" <<  element->NodeId;
+			browsedNodes.erase(element->NodeId);
+			browsedSimpleNodes.erase(element->NodeId);
+
+		}
+		void DashboardClient::deleteAndUnsubscribeNode(const ModelOpcUa::SimpleNode simpleNode) {
+			const std::list<std::shared_ptr<const ModelOpcUa::Node>> children = simpleNode.ChildNodes;
+			for(auto it = children.begin(); it != children.end();it++) {
+				std::shared_ptr<const ModelOpcUa::Node> node = *it;
+				std::shared_ptr<const ModelOpcUa::SimpleNode> simpleNode = std::dynamic_pointer_cast<const ModelOpcUa::SimpleNode>(node);
+				if(simpleNode != nullptr) {
+					const ModelOpcUa::SimpleNode simpleN = *simpleNode;
+					deleteAndUnsubscribeNode(simpleN);
+				} else {
+					std::shared_ptr<const ModelOpcUa::PlaceholderNode> placeHolderNode = std::dynamic_pointer_cast<const ModelOpcUa::PlaceholderNode>(node);
+					if(placeHolderNode != nullptr) {
+						std::list<ModelOpcUa::PlaceholderElement> instances = placeHolderNode->getInstances();
+						for(std::list<ModelOpcUa::PlaceholderElement>::iterator it = instances.begin(); it != instances.end(); it++) {
+							deleteAndUnsubscribeNode(*it);
+						}
+
+					}
+				}
+			}
+			LOG(INFO) << "Erase NodeId :" << simpleNode.NodeId;
+			browsedNodes.erase(simpleNode.NodeId);
+			browsedSimpleNodes.erase(simpleNode.NodeId);
+		}
+		void DashboardClient::subscribeEvents() {
+			auto ecbf = [](IDashboardDataClient::StructureChangeEvent sce, void* context) {
+				DashboardClient* pDashboardClient = static_cast<DashboardClient*> (context);
+				if(pDashboardClient != nullptr) {
```

Comment:
Invert the condition and return directly. So less indentations are required.

Reply:
Done.